File: solutions/2373.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n = grid.size();
        int m = n - 2;
        vector<vector<int>> res(m, vector<int>(m, 0));
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < m; ++j) {
                int current = 0;
                for (int a = -1; a <= 1; ++a) {
                    for (int b = -1; b <= 1; ++b) {
                        current = max(current, grid[i + a + 1][j + b + 1]);
                    }
                }
                res[i][j] = current;
            }
        }
        return res;
    }
};

File: rust_solutions/2373.rs
impl Solution {
    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let (m, n) = (grid.len(), grid[0].len());
        let mut ret = vec![vec![0; n - 2]; m - 2];
        
        fn find(grid: &Vec<Vec<i32>>, i: usize, j: usize) -> i32 {
            let mut ret = grid[i][j];
            for r in [i - 1, i, i + 1] {
                for c in [j - 1, j, j + 1] {
                    ret = ret.max(grid[r][c]);
                }
            }
            return ret;
        }
        for i in 0..m - 2 {
            for j in 0..n - 2 {
                ret[i][j] = find(&grid, i + 1, j + 1);
            }
        }
        ret
    }
}